### DISCUSSION TITLE
Feature request: add new commands

### DISCUSSION THREAD

**Discussion by ghost** [microsoft/vcpkg#19064] [Ideas]
Homebrew, the OSX package manager, is curated similar to vcpkg. However, Homebrew has several commands that simplify and improve the user experience:

1. The "tap" command allows using formulae not in the core repository. It actually does a "git clone" of the repository and then copies the formulae into its folders. What that would look like for vcpkg would be "git clone [repository passed from command line]" and then "cp -r repository_folder/* vcpkg/ports" (and maybe some sanity checks).
2. The "update" command simply does "git pull" in the vcpkg folder.

**Comment by traversaro** [comment]
Related issue for the first point: https://github.com/Microsoft/vcpkg/issues/114 .

**Comment by ghost** [comment]
The update command seems to be implemented already, so this is a complete duplicate.

**Comment by ghost** [comment]
Reopening because `vcpkg update` does not do a `git pull` and I think it should.

**Comment by Farwaykorse** [comment]
When `git pull` gets automated, consider adding `git gc --auto` ([documentation](https://git-scm.com/docs/git-gc)). It cleans and compresses the git repository, and is something you would regularly run on your own repositories.

The difference isn't large, only a few MBs in file-size after a year of use.
But, I think it is an easy addition, in line with the idea of user friendly operation.
And once it is in, it will stay in effect, even if vcpkg becomes an automated part of VS or other tools.
